hbase-server/src/main/java/org/apache/hadoop/hbase/security/provider/SaslServerAuthenticationProvider.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hbase.security.provider;
+
+import java.io.IOException;
+import java.util.Map;
+
+import org.apache.hadoop.hbase.HBaseInterfaceAudience;
+import org.apache.hadoop.security.UserGroupInformation;
+import org.apache.hadoop.security.token.SecretManager;
+import org.apache.hadoop.security.token.TokenIdentifier;
+import org.apache.yetus.audience.InterfaceAudience;
+import org.apache.yetus.audience.InterfaceStability;
+
+/**
+ * Encapsulates the server-side logic to authenticate a client over SASL. Tied one-to-one to
+ * a single client authentication implementation.
+ */
+@InterfaceAudience.LimitedPrivate(HBaseInterfaceAudience.AUTHENTICATION)
+@InterfaceStability.Evolving
+public interface SaslServerAuthenticationProvider extends SaslAuthenticationProvider {
```

Comment:
Sorry to park back on this @joshelser , just a minor suggestion I have while playing with implementing some providers: Might be worth define an _init_ method with a _Configuration_ parameter, for cases where custom providers could benefit from customizable properties. We could make it default in this interface, then call it from _SaslServerAuthenticationProviders.addExtraProviders_.

Reply:
> Might be worth define an init method with a Configuration parameter, for cases where custom providers could benefit from customizable properties. We could make it default in this interface, then call it from SaslServerAuthenticationProviders.addExtraProviders

That seems reasonable and making it `default` removes any pain for the implementor (should they not need it).